godot-ffi/src/lib.rs

```diff
@@ -307,29 +374,41 @@ pub unsafe fn load_class_method_table(api_level: ClassApiLevel) {
 
 /// # Safety
 ///
-/// Must be accessed from the main thread, and the interface must have been initialized.
-/// `tag_string` must be a valid type pointer of a `String` instance.
+/// - Must be accessed from the main thread.
+/// - The interface must have been initialized.
+/// - The `Scene` api level must have been initialized.
+/// - `os_class_sname` must be a valid `StringName` pointer.
+/// - `tag_string` must be a valid type pointer of a `String` instance.
 #[inline]
 pub unsafe fn godot_has_feature(
     os_class_sname: GDExtensionConstStringNamePtr,
     tag_string: GDExtensionConstTypePtr,
 ) -> bool {
     // Issue a raw C call to OS.has_feature(tag_string).
 
-    let method_bind = class_scene_api().os__has_feature();
-    let get_singleton = get_interface().global_get_singleton.unwrap();
-    let class_ptrcall = get_interface().object_method_bind_ptrcall.unwrap();
+    // SAFETY: Called from main thread, interface has been initialized, and the scene api has been initialized.
+    let method_bind = unsafe { class_scene_api() }.os__has_feature();
+
+    // SAFETY: Called from main thread, and interface has been initialized.
+    let interface = unsafe { get_interface() };
+    let get_singleton = interface.global_get_singleton.unwrap();
+    let class_ptrcall = interface.object_method_bind_ptrcall.unwrap();
 
-    let object_ptr = get_singleton(os_class_sname);
+    // SAFETY: Interface has been initialized, and `Scene` has been initialized, so `get_singleton` can be called. `os_class_sname` is a valid
+    // `StringName` pointer.
+    let object_ptr = unsafe { get_singleton(os_class_sname) };
     let mut return_ptr = false;
     let type_ptrs = [tag_string];
 
-    class_ptrcall(
-        method_bind.0,
-        object_ptr,
-        type_ptrs.as_ptr(),
-        return_ptr.sys_mut(),
-    );
+    // SAFETY: We are properly passing arguments to make a ptrcall.
+    unsafe {
+        class_ptrcall(
+            method_bind.0,
+            object_ptr,
+            type_ptrs.as_ptr(),
+            return_ptr.sys_mut(),
+        )
```

Comment:
This kind of `// SAFETY` statement is not very expressive -- it may be a symptom of forcing to write `// SAFETY` above every `unsafe` block. Similar to situations where we just call Godot APIs.

Maybe it's better to omit the comment in such cases? :slightly_smiling_face: